Use RTC to set system clock on boot

WaterByWind opened the pull request:
Summary:  Enable kernel use of RTC to set system clock early in boot.

For some reason the pine64 is one of very few configs that do not have RTC_HCTOSYS enabled, but that config also has an erroneous RTC_HCTOSYS_DEVICE configured.

The NEXT and DEV branches both are also missing RTC_HCTOSYS for sunxi-64, while most of the other configs do have this enabled.  It should not be an issue to enable this kernel config here as well for those boards (including pine64) that do have an RTC.

WaterByWind commented:
I've only provided one commit for DEV branch, as this is a feature change.  I can also add a commit for NEXT if desired.

This has been part of my standard config when I build custom kernels for quite some time and haven't ever seen an issue.

lanefu commented:
Might as well do next and dev.

What about 32bit sunxi kernels?  Or is this really just related to H6 boards?

WaterByWind commented:
This is one of those options that typically is left enabled by default generically unless there is a specific reason to _not_ enable it.  The feature code (drivers/rtc/hctosys.c) is very simple but does handle exceptions (IE: no RTC present) properly.

I just don't have a way to do any boot-time validation but could add to the 32-bit -sunxi configs as well.

WaterByWind commented:
I've added commits to cover the additional three kernel configs.  32 & 64 bit, NEXT and DEV are all now updated.  I configured and built 32-bit but did not have any way to validate boot.

This config symbol is a default 'y' in mainline Kconfig for the rtc driver and this PR brings these four configurations inline with both the rest of the supported boards and the mainline default.
